### PR TITLE
Change IndexerZkConfig to use @JacksonInject instead of just straight @Inject

### PR DESCRIPTION
**Pull request by drcrallen**
Also added more unit tests and changed them so they don't clobber `System.getProperties()`

### CODE REVIEW

**Comment by drcrallen** [comment]
little more testing needed.

**Comment by drcrallen** [comment]
Ok it was fine.

**Comment by drcrallen** [comment]
Problem was that JacksonInject resolves at bind time rather than inject time, which is kind of annoying for testing purposes.

That was for a one-off test though, and the test proposed in this PR is a way of "faking" dynamic injection.

File: indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java [hunk 1]
```diff
@@ -29,8 +30,8 @@
  */
 public class IndexerZkConfig
 {
-  @Inject
   @JsonIgnore
+  @JacksonInject
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java]: Can we create a `@JsonCreator` constructor that sets all the properties and mark them as final in the class? That will make it more obvious what is injected and what not.

Comment (xvrl) [inline indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java]: we can also remove the lone setter once we've done that.

**Comment by drcrallen** [comment]
Believed to fix #1020

File: indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java (hunk 1 above)
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java]: https://github.com/druid-io/druid-api/blob/master/src/main/java/io/druid/guice/JsonConfigurator.java#L148

Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/server/initialization/IndexerZkConfig.java]: Fixed

**Comment by drcrallen** [comment]
I tested this with a remote tasking Overlord and it seems to get `druid.zk.paths.base` into the overlord's index zookeeper config correctly.

**Pull request merged**